### PR TITLE
SW-5122 Add ability to style the Select component

### PR DESCRIPTION
**Pull request by nickgraz**
- Add select styling props since useStyles is deprecated and we need the ability to control the styling for the participant project dropdown

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#510** <a href="https://app.graphite.dev/github/pr/terraware/web-components/510?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/Select/SelectT.tsx [hunk 1]
```diff
@@ -7,59 +7,67 @@ import Icon from '../Icon/Icon';
...
   renderOption: (option: T) => React.ReactNode;
+  required?: boolean;
+  selectedValue?: T;
+  selectStyles?: SelectStyles;
   toT: (input: string) => T;
-  displayLabel: (option: any) => string;
   tooltipTitle?: TooltipProps['title'];
-  editable?: boolean;
-  required?: boolean;
+  warningText?: string | string[];
```
Comment (nickgraz) [inline src/components/Select/SelectT.tsx:41]: Besides the addition of `selectStyles?: SelectStyles;`, this is just an alphabetical reordering

File: src/components/Select/SelectT.tsx [hunk 2]
```diff
@@ -7,59 +7,67 @@ import Icon from '../Icon/Icon';
...
-    toT,
-    displayLabel,
-    tooltipTitle,
-    editable,
     required,
+    selectedValue,
+    selectStyles,
+    tooltipTitle,
+    toT,
+    warningText,
```
Comment (nickgraz) [inline src/components/Select/SelectT.tsx:70]: Besides the addition of `selectStyles`, this is just an alphabetical reordering

File: src/components/Select/SelectT.tsx [hunk 3]
```diff
@@ -81,28 +89,31 @@ export default function SelectT<T>(props: SelectTProps<T>): JSX.Element {
...
+      } else if (fixedMenu) {
+        setFixedMenuPositioned(false);
       }
-    } else if (fixedMenu) {
-      setFixedMenuPositioned(false);
-    }
-  }, [fixedMenu, openedOptions, hasOptions]);
+    },
+    [fixedMenu, openedOptions, hasOptions]
+  );
```
Comment (nickgraz) [inline src/components/Select/SelectT.tsx:116]: These seem to be `prettier` changes, nothing was modified in here.

**Comment by nickgraz** [comment]
publish rc

**Comment by nickgraz** [comment]
publish rc

File: src/components/Select/SelectT.tsx [hunk 4]
```diff
@@ -226,7 +236,13 @@ export default function SelectT<T>(props: SelectTProps<T>): JSX.Element {
         </label>
       )}
       <div className={`textfield-container ${fullWidth ? 'textfield-container--fullWidth' : ''}`}>
-        <div id={id} className={selectClass} onClick={toggleOptions} ref={inputRef}>
+        <div
+          id={id}
+          className={selectClass}
+          onClick={toggleOptions}
+          ref={inputRef}
+          style={(selectStyles || {}).inputContainer}
```
Comment (constanzauanini) [inline src/components/Select/SelectT.tsx:244]: Accessing inputContainer from {} is not getting an error?

**Comment by nickgraz** [comment]
Converting to a draft because there are other style props that need to be implemented

**Comment by nickgraz** [comment]
@constanzauanini ready to go again!

**Comment by nickgraz** [comment]
publish rc

**Comment by nickgraz** [comment]
publish rc

**Comment by nickgraz** [comment]
publish rc

**Comment by nickgraz** [comment]
publish rc

File: src/components/Icon/Icon.tsx [hunk 5]
```diff
@@ -17,15 +17,21 @@ const useStyles = makeStyles(() => ({
 }));
 
 export interface Props {
-  name: IconName;
-  size?: Size;
   className?: string;
   fillColor?: string;
+  name: IconName;
+  size?: Size;
+  style?: Record<string, unknown>;
```
Comment (constanzauanini) [inline src/components/Icon/Icon.tsx:24]: I did this same change on my react 18 branch, but I used type CSSProperties

**Review by constanzauanini — APPROVED** [review]
Looks good, just one comment

**Pull request merged**